### ISSUE TITLE
Optimise Webpack Bundling & Polyfills

### ISSUE THREAD

**Issue by sirockin** [json-editor/json-editor#488]
### General information

#486 implements Webpack but does not optimise bundling or polyfills.

Currently suggested optimisations are:
1. Lazy-load all themes other than html
1. Lazy-load all editors
1. Remove existing ie9 polyfill and aim to replace with standard Webpack polyfill if necessary

To be done once #486 is merged with development/2.x

**Comment by pmk65** [comment]
@sirockin I can make a list of the mandatory editors (no lazy loading) and the optional editors (lazy loading) if you need it?
Also should we move the resolvers to separate files like in this [PR](https://github.com/jdorn/json-editor/pull/322/commits/7e9dcbfa31bb13c9f499ad56c0fd57a56a486812)

https://github.com/jdorn/json-editor/blob/7e9dcbfa31bb13c9f499ad56c0fd57a56a486812/src/resolvers/describedby.js#L1-L11

**Comment by sirockin** [comment]
@pmk65:
> @sirockin I can make a list of the mandatory editors (no lazy loading) and the optional editors (lazy loading) if you need it?
Yes that would be great, thanks!

> Also should we move the resolvers to separate files like in this

Okay will look into that too

**Comment by pmk65** [comment]
***Mandatory Editors***
array.js
base64.js _(Not sure about this one.)_
checkbox.js
enum.js
hidden.js
integer.js
multiple.js
multiselect.js
null.js
number.js
object.js
select.js
string.js


***Optional Editors***
ace.js
autocomplete.js
button.js
choices.js
datetime.js
describedby.js
info.js
jodit.js
radio.js _(Might be added to mandatory)_
sceditor.js
table.js _(Might be added to mandatory)_
upload.js
uuid.js
ip.js
select2.js
selectize.js
signature.js
simplemde.js
starrating.js
array\choices.js
array\select2.js
array\selectize.js

**Comment by sirockin** [comment]
I've done some work on this PR but the changes required are fairly complex since 
1. Lazy loading requires use of async/await constructs using ES2016
1. Adding polyfills for the above adds quite a lot of extra bytes to the output
1. Using ES2016 requires transpilation which may make even unminified code harder to read
1. Even the relatively simple bit - loading themes asynchonously - needs some work to ensure that late evaluation doesn't break tests
1. I'm still not completely up to speed with the lifecycle of the various components

For this reason I'm putting this work on hold until we're all happy with the general state of the webpack setup.

**Comment by pmk65** [comment]
@sirockin @schmunk42 Should we release a 2.x beta version with the current changes, or should we continue working on it without a beta?

**Comment by schmunk42** [comment]
We could create an `alpha`...

**Comment by sirockin** [comment]
I'd definitely be up for publishing something - an `alpha` sounds good.

**Comment by pmk65** [comment]
Let's release an `alpha` version then. 👍

**Comment by pmk65** [comment]
> 
> 
> I've done some work on this PR but the changes required are fairly complex since
> 
>     1. Lazy loading requires use of async/await constructs using ES2016
> 
>     2. Adding polyfills for the above adds quite a lot of extra bytes to the output
> 
>     3. Using ES2016 requires transpilation which may make even unminified code harder to read
> 
>     4. Even the relatively simple bit - loading themes asynchonously - needs some work to ensure that late evaluation doesn't break tests
> 
>     5. I'm still not completely up to speed with the lifecycle of the various components
> 
> 
> For this reason I'm putting this work on hold until we're all happy with the general state of the webpack setup.

@sirockin So what's the next step, now we have an alpha version? Transpile code to ES2016?
